lib/spack/spack/multimethod.py

```diff
@@ -34,6 +35,23 @@
 from spack.spec import Spec
 
 
+class MultiMethodMeta(type):
+    """This allows us to track the class's dict during instantiation."""
+
+    #: saved dictionary of attrs on the class being constructed
+    _locals = None
+
+    @classmethod
+    def __prepare__(cls, name, bases, **kwargs):
+        """Save the dictionary that will be used for the class namespace."""
+        MultiMethodMeta._locals = dict()
+        return MultiMethodMeta._locals
+
+    def __init__(cls, name, bases, attr_dict):
+        """Clear out the cached locals dict once the class is built."""
+        MultiMethodMeta._locals = None
```

Comment:
This method should call `super` at the end, otherwise it will interrupt the MRO chain and its use as a mixin will depend on ordering within the base classes.

Reply:
fixed!  nice catch.